File: src/main/java/io/github/takenoha/towerdefense/persistence/VoucherRedeemResult.java
package io.github.takenoha.towerdefense.persistence;

import java.util.Objects;

/** Voucher plus its durable redeem operation after a reservation read. */
public record VoucherRedeemResult(
        OperationOutcome outcome,
        ResourceVoucher voucher,
        VoucherRedeemOperation operation) {
    public VoucherRedeemResult {
        Objects.requireNonNull(outcome, "outcome");
        Objects.requireNonNull(voucher, "voucher");
        Objects.requireNonNull(operation, "operation");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/persistence/VoucherRedeemResult.kt
package io.github.takenoha.towerdefense.persistence

import java.util.Objects
import kotlin.jvm.JvmRecord

/** Voucher plus its durable redeem operation after a reservation read. */
@JvmRecord
data class VoucherRedeemResult(
    val outcome: OperationOutcome,
    val voucher: ResourceVoucher,
    val operation: VoucherRedeemOperation,
) {
    init {
        Objects.requireNonNull(outcome, "outcome")
        Objects.requireNonNull(voucher, "voucher")
        Objects.requireNonNull(operation, "operation")
    }
}
